packages/lodestar/src/network/network.ts

```diff
@@ -132,25 +137,24 @@ export class Libp2pNetwork extends (EventEmitter as {new (): NetworkEventEmitter
    * @param subnet the subnet calculated from committee index
    * @param inPeerIds peers already have this subnet
    */
-  private async connectToNewPeersBySubnet(subnet: number, inPeerIds: string[] = []): Promise<number> {
+  public async connectToNewPeersBySubnet(subnet: number, inPeerIds: string[] = []): Promise<number> {
     const discv5Peers = (await this.searchDiscv5Peers(subnet)) || [];
     const peerIds = discv5Peers.filter((peerId) => !inPeerIds.includes(peerId.toB58String()));
-    // make sure they still connect to same subnet
     let count = 0;
+    // make sure they still connect to same subnet
     for (const peerId of peerIds) {
-      // we'll dial thru sendRequest so don't need to do connect() like in the spec
+      // will automatically get metadata once we connect
       try {
-        const metadata = await this.reqResp.metadata(peerId);
-        if (metadata === null) throw Error("No metadata");
-        if (metadata.attnets[subnet]) {
-          count++;
+        await this.connect(peerId);
+        count++;
+        if (inPeerIds.length + count >= MAX_NUM_PEERS_PER_SUBNET) {
+          break;
         }
-      } catch (err) {
-        this.logger.warn(`Cannot get metadata from ${peerId.toB58String()}: ${err.message}`);
-      }
-      if (count < 10) {
-        // TODO: decide max peers per subnet to connect?
-        break;
+      } catch (e) {
+        this.logger.warn(
+          `Cannot connect to peer ${peerId.toB58String()} for subnet ${subnet}`,
+          e.message.substring(0, 30)
```

Comment:
there are a lot of error, one of them is `onAbort` and it appears in multilines, also we run this task frequently for all missing subnets
can we go like this and monitor it for a while, will change to display more if we think it's valuable to do so?

Reply:
hmm, I think on Abort can be overriden?
I'm afraid if we receive some unique error it would be too late to change, rather leave it multilined until we figure out fix for it